File: ruby/4.rb
# project euler 4: find the largest palindrome made from the product of two 3-digit numbers
t = Time.new.to_f
def palindrome?(n)
	n.to_s == n.to_s.reverse
end
largest = 0
for i in 0..999
	for j in 0..999
		p = i*j
		largest = p if palindrome?(p)&&p>largest
	end
end
puts "result: #{largest} - found in #{((Time.new.to_f - t)*1000).round 3} ms"

File: lua/4.lua
--project euler 4: find the largest palindrome made from the product of two 3-digit numbers
package.path = "..\\libraries\\?.lua;" .. package.path
require "lib"
local t = os.clock()

local highest = 0

for n1=100,999 do
    for n2=100,999 do
        local n = n2*n1
        if (lib.is_palindrome(n)) and (n > highest) then
            highest = n
        end
    end
end

print("result: "..highest.." - found in "..(os.clock() - t).." s")